client/state/ui/drop-zone/reducer.js

```diff
@@ -17,10 +13,16 @@ import {
 
 import { createReducer } from 'state/utils';
 
-const isVisible = createReducer( false,
+const isVisible = createReducer( {},
 	{
-		[ DROPZONE_SHOW ]: stubTrue,
-		[ DROPZONE_HIDE ]: stubFalse,
+		[ DROPZONE_SHOW ]: ( state, { dropZoneName } ) => ( {
+			...state,
+			[ dropZoneName ]: true,
+		} ),
+		[ DROPZONE_HIDE ]: ( state, { dropZoneName } ) => ( {
+			...state,
+			[ dropZoneName ]: false,
+		} ),
```

Comment:
I tried using `keyedReducer` but for some reason it totally ignores the second reducer ( `hide` ) and only the first one works.

I'd be happy to circle around this code in the future and improve it if we can figure out the bug, but for now I think this implementation will work fine! :) 

I left a note in the code to point to that possible improvement.